proxy/clients/clients.go

```diff
@@ -102,11 +108,36 @@ func NewGRPCClientConn(opts *Options) (*grpc.ClientConn, error) {
 			return nil, err
 		}
 	}
-	return grpc.Dial(
-		connAddr,
+
+	dialOptions := []grpc.DialOption{
 		grpc.WithTransportCredentials(cert),
 		grpc.WithPerRPCCredentials(grpcAuth),
-		grpc.WithUnaryInterceptor(metrics.GRPCClientInterceptor("proxy")),
+		grpc.WithChainUnaryInterceptor(metrics.GRPCClientInterceptor("proxy"), grpcTimeoutInterceptor(opts.RequestTimeout)),
 		grpc.WithStatsHandler(&ocgrpc.ClientHandler{}),
+		grpc.WithDefaultCallOptions([]grpc.CallOption{grpc.WaitForReady(true)}...),
+	}
+
+	if opts.ClientDNSRoundRobin {
+		dialOptions = append(dialOptions, grpc.WithBalancerName(roundrobin.Name))
```

Comment:
So this is a bit annoying of gRPC.

`WithBalancerName`
>  Deprecated: use WithDefaultServiceConfig and WithDisableServiceConfig instead. Will be removed in a future 1.x release.
https://godoc.org/google.golang.org/grpc#WithBalancerName




`WithDefaultServiceConfig`
> This API is EXPERIMENTAL.
https://godoc.org/google.golang.org/grpc#WithDefaultServiceConfig


> Package balancer defines APIs for load balancing in gRPC. All APIs in this package are experimental.
> https://godoc.org/google.golang.org/grpc/balancer

Reply:
Yes.  I wouldn't mind going with experimental if it appeared to be usable at all.  I don't see enough documentation or even code to actually set up a dns lookaside client with ServiceConfig right now.  I doubt WithBalancerXXXX will be removed any time soon.